src/source/mod.rs

```diff
@@ -246,6 +248,20 @@ where
         fadein::fadein(self, duration)
     }
 
+    /// Applies a linear gain ramp to the sound
+    #[inline]
+    fn linear_gain_ramp(
```

Comment:
Would you like `fade_in()` to return a `LinearGainRamp<Self>` or do we want to erase its type somehow?

Reply:
You know what lemmie crunch on this a bit more.

Reply:
This is done.

Reply:
good solution, since FadeIn still returns a FadeIn nothing changed for anyone using FadeIn and the introduction of the linear gain ramp is non-breaking :tada: